File: spigot/src/main/java/me/devsnox/pingcounter/spigot/PingListener.java
package me.devsnox.pingcounter.spigot;

import me.devsnox.pingcounter.core.api.CounterEngine;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.server.ServerListPingEvent;

public class PingListener implements Listener {

    private final CounterEngine engine;

    public PingListener(final CounterEngine engine) {
        this.engine = engine;
    }

    @EventHandler
    public void onPing(final ServerListPingEvent event) {
        this.engine.addPing(event.getAddress());
    }
}

File: engine-modules/velocity-engine/src/main/java/me/devsnox/pincounter/engine/velocity/PingListener.kt
package me.devsnox.pincounter.engine.velocity

import com.velocitypowered.api.event.Subscribe
import com.velocitypowered.api.event.proxy.ProxyPingEvent
import me.devsnox.pingcounter.api.CounterEngine

class PingListener internal constructor(private val engine: CounterEngine) {

    @Subscribe
    fun onPing(event: ProxyPingEvent) {
        this.engine.addPing(event.connection.remoteAddress.address)
    }
}
